chapter-5/Hangman/GameHangman/GameHangman.cpp

```diff
@@ -0,0 +1,179 @@
+﻿#include <iostream>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <ctime>
+#include <cctype>
+using namespace std;
+
+/*
+Псевдокод
+	Функция rn_seed_gen()
+	{
+		Запускает повсев случайных чисел
+	}
+
+	Функция get_words()
+	{
+		Создать вектор 
+		Заности слова в вектор 
+		Перемешивает порядок слов в векторе
+		Вернуть вектор		
+	}
+
+	Функция get_init_soFar(слово)
+	{
+		Создать строку, состоящую из "-" длинной, равной длинне слова
+		Вернуть строку
+	}
+
+	Функция display_game_stat(soFar, использованные_буквы, количество_ошибок)
+	{
+		Вывести количество оставшихся попыток
+		Вывести использованные буквы
+		Вывести текущее состояние угадываемого слова
+	}
+
+	Функция get_user_var(использованные_буквы)
+	{
+		Получить ввод от пользователя
+		Преобразовать букву в верхний регистр
+		Пока введенная буква уже использовалась:
+			сообщить пользователю, что эта буква уже использовалась
+			получить ввод буквы от пользователя
+			преобразовать букву в верхний регистр
+		Вернуть введенную букву
+	}
+
+	Функция get_upd_SoFar(слово, soFar, буква)
+	{
+		Создать новую строку newSoFar, копию soFar
+		Сля каждого символа в слове:
+			Если символ совпадает с буквой:
+				заменить соответствующий символ в newSoFar на букву
+		Вернуть newSoFar
+	}
+
+	Функция game_start(слово)
+	{
+		Инициализировать soFar вызовом get_init_soFar(слово)
+		Инициализировать использованные_буквы пустой строкой
+		Инициализировать количество_ошибок нулем
+		Пока количество_ошибок меньше MAX_WRONG и soFar не равна слову:
+			вывести состояние игры вызовом display_game_stat
+			получить букву от пользователя вызовом get_user_var
+			добавить букву в использованные_буквы
+			Если буква присутствует в слове:
+				сообщить, что буква верная
+				обновить soFar вызовом get_upd_SoFar
+			Иначе:
+				сообщить, что буква неверная
+				увеличить количество_ошибок
+		Если количество_ошибок равно MAX_WRONG:
+			вывести сообщение о проигрыше
+		Иначе:
+			вывести сообщение о выигрыше
+		Вывести загаданное слово
+	}
+*/
+
+void rn_seed_gen()
+{
+	srand(static_cast<unsigned int>(time(0)));
+}
+const int МАХ_WRONG = 8;
+
+vector<string> get_words()
+{
+	vector<string>words;
+	words.push_back("GUESS");
+	words.push_back("HANGMAN");
+	words.push_back("DIFFICULT");
+	random_shuffle(words.begin(), words.end());
+	rn_seed_gen();
```

Comment:
Нужно изменить порядок - сначала вызывать seed, а потом shuffle:
```suggestion
	rn_seed_gen();
	random_shuffle(words.begin(), words.end());
```